logic/service/blueprint-data-mapping.js

```diff
@@ -0,0 +1,140 @@
+var DataMapping = require("logic/service/data-mapping").DataMapping;
+
+/**
+ * Maps raw data to data objects of a specific type by using a blueprint.
+ **
+ * @class
+ * @extends external:Montage
+ */
+exports.BlueprintDataMapping = DataMapping.specialize(/** @lends BlueprintDataMapping.prototype */ {
+
+    _blueprint: {
+        value: undefined
+    },
+
+    /***************************************************************************
+     * Mapping
+     */
+
+    /**
+     * Convert raw data to data objects of an appropriate type.
+     *
+     * Subclasses should override this method to map properties of the raw data
+     * to data objects, as in the following:
+     *
+     *     mapRawDataToObject: {
+     *         value: function (object, data) {
+     *             object.firstName = data.GIVEN_NAME;
+     *             object.lastName = data.FAMILY_NAME;
+     *         }
+     *     }
+     *
+     * The default implementation of this method copies the properties defined
+     * by the raw data object to the data object.
+     *
+     * @method
+     * @argument {Object} object - An object whose properties must be set or
+     *                             modified to represent the raw data.
+     * @argument {Object} data   - An object whose properties' values hold
+     *                             the raw data.
+     * @argument {?} context     - A value that was passed in to the
+     *                             [addRawData()]{@link DataService#addRawData}
+     *                             call that invoked this method.
+     */
+    mapRawDataToObject: {
+        value: function (data, object, context) {
```

Comment:
We should expect that the raw data is described by it's own ObjectDescriptor), and the object as well. Assuming the mapping has an index for each side of the conversion, as we loop on properties of one side we should be able to find corresponding "bindings" to apply. Or maybe there's a list of mapping/bindings for each side (the same if two-way) and that's what's iterated on. Because you could have more mapping expressions/bindings on one side than the  other